partner_communication_switzerland/migrations/10.0.2.4.6/post-migration.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import fields
+from datetime import datetime, timedelta
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    if not version:
+        return
+
+    sponsorships = env['recurring.contract'].search([
+        ('activation_date', '!=', None)
+    ])
+    for sponsorship in sponsorships:
+        # if activation date is more than 24h old
+        if sponsorship.activation_date <= fields.Datetime.to_string(
```

Comment:
Here you compare strings. It's safer to compare date objects to determine which is older. You can use `fields.Datetime.from_string(sponsorship.activation_date)`